pogom/search.py

```
@@ -203,18 +231,67 @@ def any_spawnpoints_in_range(coords):
+    global spawns, Shash, going
+    log.info('Search ss overseer starting')
+    search_items_queue = Queue()
+    parse_lock = Lock()
+
+    # Create a search_worker_thread per account
+    log.info('Starting search worker threads')
+    for i, account in enumerate(args.accounts):
+        log.debug('Starting search worker thread %d for user %s', i, account['username'])
+        t = Thread(target=search_worker_thread_ss,
+                   name='ss_search_worker_{}'.format(i),
+                   args=(args, account, search_items_queue, parse_lock, encryption_lib_path))
+        t.daemon = True
+        t.start()
+
+    try:
+        with open(args.spawnpoint_scanning) as file:
+            try:
+                spawns = json.load(file)
+            except ValueError:
+                log.error(args.spawnpoint_scanning + " is not valid")
+                return
+            file.close()
+    except IOError:
+        log.error("Error opening " + args.spawnpoint_scanning)
+        return
+
+    for spawn in spawns:
+        Shash[spawn['lng']] = spawn['time']
+    # sort spawn points
+    spawns.sort(key=itemgetter('time'))
+    log.info('Total of %d spawns to track', len(spawns))
+    # find start position
+    pos = SbSearch(spawns, (curSec() + 3540) % 3600)
+    while True:
+        while timeDif(curSec(), spawns[pos]['time']) < 60:
+            time.sleep(1)
+        location = []
+        location.append(spawns[pos]['lat'])
+        location.append(spawns[pos]['lng'])
+        location.append(0)
+        for step, step_location in enumerate(generate_location_steps(location, 1), 1):
+                log.debug('Queueing step %d @ %f/%f/%f', pos, step_location[0], step_location[1], step_location[2])
+                search_args = (step, step_location, spawns[pos]['time'])
+                search_items_queue.put(search_args)
+        pos = (pos + 1) % len(spawns)
+        if pos == 0:
```

why is this section here?

if the search queue is long it can cause many spawns at the beginning of the list to be silently skipped, and if the search queue is short it can cause it to re-scan the end of the list